distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date_Range.enso

```diff
@@ -0,0 +1,484 @@
+
+    ## Applies a function to each element in the range, producing a vector of
+       results.
+
+       Arguments:
+       - function: The function to apply to each date in the range.
+
+       > Example
+         Create a vector that contains the numbers twice that of the numbers in
+         the range.
+
+             1.up_to 10 . map (*2)
+    map : (Date -> Any) -> Vector Any
+    map self function =
+        Vector.new self.length (i -> function (self.internal_at i))
+
+    ## Converts the range to a vector containing the dates in the range.
+
+       > Example
+         Getting a vector of dates from 2021-05-07 to 2021-05-10 (exclusive).
+
+             (Date.new 2021 05 07).up_to (Date.new 2021 05 10) . to_vector
+    to_vector : Vector Date
+    to_vector self = self.map x->x
+
+    ## Checks if this range is empty.
+
+       > Example
+         Checking if the range of days from start of 2020 to start of 2023 is empty.
+
+             (Date.new 2020).up_to (Date.new 2023) . is_empty
+    is_empty : Boolean
+    is_empty self = self.length == 0
+
+    ## Checks if this range is not empty.
+
+       > Example
+         Checking if the range of days from start of 2020 to start of 2023 is not empty.
+
+             (Date.new 2020).up_to (Date.new 2023) . is_empty
+    not_empty : Boolean
+    not_empty self = self.is_empty.not
+
+    ## Returns a vector of all elements of this range which satisfy a condition.
+
+       Arguments:
+       - filter: The filter to apply to the range. It can either be an instance
+         of `Filter_Condition` or a predicate taking a value and returning a
+         boolean value indicating whether the corresponding element should be
+         kept or not.
+
+       > Example
+         Selecting all elements that are greater than 2020-10-12.
+
+             (Date.new 2020 10 01).up_to (Date.new 2020 10 15) . filter (> (Date.new 2020 10 12))
+             (Date.new 2020 10 01).up_to (Date.new 2020 10 15) . filter (Filter_Condition.Greater than=(Date.new 2020 10 12))
+    filter : (Filter_Condition | (Date -> Boolean)) -> Vector Date
+    filter self filter = case filter of
+        _ : Filter_Condition -> self.filter filter.to_predicate
+        predicate : Function ->
+            builder = self.fold Vector.new_builder builder-> elem->
+                if predicate elem then builder.append elem else builder
+            builder.to_vector
+
+    ## Applies a function for each element in the range.
+
+       Arguments:
+       - function: The function to apply to each integer in the range.
+
+       > Example
+         To print all dates from 2020-10-01 to 2020-10-05.
+             (Date.new 2020 10 01).up_to (Date.new 2020 10 05) include_end=True . each IO.println
+    each : (Date -> Any) -> Nothing
+    each self function =
+        (0.up_to self.length).each ix->
+            function (self.internal_at ix)
+
+    ## PRIVATE
+       Applies a function to each element of the range.
+
```

Comment:
Why is this one PRIVATE? 
Guess question for `Range` too !

Reply:
Yeah, I kept what was there in Range for consistency.

Hmm, I think it may have been added because we needed it for some internal function and didn't want to expose it. But I think it's definitely fair to change it.

----------------

OTOH I'm wondering if all of `each` on our collections should be visible in CB? Maybe we should mark them as PRIVATE+ADVANCED for now. That's because in the GUI, you probably almost always want to use `map` instead - as we practically always want to see the result.

For example, the one use-case I see for `each` is to write _every_ element in some collection to a separate file. But then I think we still want to use a `map` to get the filenames of the written files. 

What do you think?

Reply:
I guess as `each` and `each_with_index` dont work in IDE `PRIVATE ADVANCED` make sense.